sdk/template/Azure.Template/src/Generated/Models/BaseClassWithDiscriminator.cs

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using Azure.Core;
+
+namespace Azure.Template.Models
+{
+    /// <summary> Example base class that has a discriminator property. </summary>
+    // Note: base classes are abstract for polymorphic types
+    public abstract partial class BaseClassWithDiscriminator : BaseClass
```

Comment:
@KrzysztofCwalina @tg-msft @lmazuel Can you confirm that base class that has discriminator property should never be instantiated?

Reply:
Management plan is doing this.  The client can still return an instance of the base class with an unknown discriminator using the "internal unknown" type, which in this PR is `DerivedFromBaseClassWithUnknownDiscriminator`

Reply:
Yes, it happens that you need to return the base class and you don't know the discriminator (sorry for the late answer)

Reply:
@lmazuel - Thanks!  Can you help us understand the scenario you're describing?